app/models/job.ts

```diff
@@ -226,6 +226,40 @@ export const statesToDefaultMessages: any = Object.values(stateMachine.states).r
 
 const defaultMessages = Object.values(statesToDefaultMessages);
 
+
+/**
+ * When the job status is updated we might want to remove some of the prior message parts
+ * while retaining some of the information from the message. For example, we might want
+ * "The job is generating a preview before auto-pausing. The CMR identified 10 granules." to become
+ * "The CMR identified 10 granules." when a job goes from previewing to running.
+ * @param message - the job message that may need changing
+ * @param partsToRemove - an array of strings to remove (sentences without periods--e.g. ["the job is paused"])
+ * @param partsFilter - an optional filter for removal of message parts that do not match this filter
+ * @returns the new message, with partsToRemove removed and sentence structure maintained,
+ * or an empty string if the new message has no retained parts
+ */
+function removeMessageParts(
+  message: string,
+  partsToRemove,
+  partsFilter: (p: string) => boolean = undefined,
+): string {
+  if (!message) {
+    return;
+  }
+  let acceptableParts = message
+    .split('.')
```

Comment:
Deciding that a '.' means the end of a sentence seems problematic, but if you're only removing a portion if it's an exact match to some message it should be fine.

Reply:
I don't want to make this more complicated than it has to be, but does it make more sense to store the message as an array (or even a map) of messages that we can more easily pick out the parts we want to filter out (or keep)? Then we just have to concatenate the message parts whenever we want to send or log the message.